src/uu/du/src/du.rs

```diff
@@ -309,7 +390,12 @@ fn du(
                 show!(
                     e.map_err_context(|| format!("cannot read directory {}", my_stat.path.quote()))
                 );
-                return Box::new(iter::once(my_stat));
+                tx.send(StatPrintInfo {
+                    stat: my_stat.clone(),
+                    depth,
+                })
+                .unwrap();
```

Comment:
maybe handle the error here ?

Reply:
Yup, I will, thanks! Like I said in my comment above, this is not really ready for review yet, I'm aware this PR is currently a complete mess and not production quality yet. I'll mark it is not a draft once it's ready for review!

Reply:
Added some basic error handling for calls to `print_tx.send`, although it probably is not as refined as it could be. Going to resolve this particular thread since this part of the code has changed a good bit since this review.